<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <title>rOpenSci | rsnps tutorial</title>

     
    <script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
    <script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>

    <script type="text/javascript">
    hljs.configure({languages: []});
    hljs.initHighlightingOnLoad();
    </script>
    <link rel="stylesheet" type="text/css" href="/css/style-new.css" />
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/manifest.json">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#4b8add">
<meta name="theme-color" content="#4b8add">


    <meta name="viewport" content="width=device-width, initial-scale=1">
</head>
<body>

    
    <div id="header">
        <div class="container">
            <div class="row center-m">
                
                <div class="col-1">
                    <a href="/index.html"><img src="/img/icon_lettering_white.svg" /></a>
                </div>
                
                
                <nav class="col-6 col-offset-4 top-3">
                    <ul class="row between">
						<li><a href="/about/">About</a></li>
                        <li><a href="/blog/">Blog</a></li>
                        <li><a href="/packages/">Packages</a></li>
                        <li><a href="/community/">Community</a></li>
                        <li><a href="http://discuss.ropensci.org/" target="_blank">Discuss</a></li>

                    </ul>
                </nav>
                
            </div>
        </div>
    </div>



    <article>
        <div class="container top-20">
            
            <div class="row centered">
                
                <div class="col-7 bottom-20 maintext">
                    <h1>rsnps tutorial</h1>
                    <br>
                    <h4><small style="color:grey">for v0.2.0</small></h4>
                    <br>

                    <article>
                    

<h3 id="installation">Installation</h3>

<p>Stable version from CRAN</p>

<pre><code class="language-r">install.packages(&quot;rsnps&quot;)
</code></pre>

<p>Or get from Github</p>

<pre><code class="language-r">install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;ropensci/rsnps&quot;)
</code></pre>

<pre><code class="language-r">library(&quot;rsnps&quot;)
</code></pre>

<h3 id="opensnp-data">OpenSNP data</h3>

<h4 id="all-genotypes">All Genotypes</h4>

<p>Get genotype data for all users at a particular SNP</p>

<pre><code class="language-r">allgensnp(snp='rs7412')[1:3]
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/rs7412.json
</code></pre>

<pre><code>#&gt; [[1]]
#&gt; [[1]]$snp
#&gt; [[1]]$snp$name
#&gt; [1] &quot;rs7412&quot;
#&gt;
#&gt; [[1]]$snp$chromosome
#&gt; [1] &quot;19&quot;
#&gt;
#&gt; [[1]]$snp$position
#&gt; [1] &quot;44908822&quot;
#&gt;
#&gt;
#&gt; [[1]]$user
#&gt; [[1]]$user$name
#&gt; [1] &quot;Luke Reid&quot;
#&gt;
#&gt; [[1]]$user$id
#&gt; [1] 6621
#&gt;
#&gt; [[1]]$user$genotypes
#&gt; [[1]]$user$genotypes[[1]]
#&gt; [[1]]$user$genotypes[[1]]$genotype_id
#&gt; [1] 5023
#&gt;
#&gt; [[1]]$user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;CC&quot;
#&gt;
#&gt;
#&gt;
#&gt;
#&gt;
#&gt; [[2]]
#&gt; [[2]]$snp
#&gt; [[2]]$snp$name
#&gt; [1] &quot;rs7412&quot;
#&gt;
#&gt; [[2]]$snp$chromosome
#&gt; [1] &quot;19&quot;
#&gt;
#&gt; [[2]]$snp$position
#&gt; [1] &quot;44908822&quot;
#&gt;
#&gt;
#&gt; [[2]]$user
#&gt; [[2]]$user$name
#&gt; [1] &quot;Ganesha18&quot;
#&gt;
#&gt; [[2]]$user$id
#&gt; [1] 6598
#&gt;
#&gt; [[2]]$user$genotypes
#&gt; [[2]]$user$genotypes[[1]]
#&gt; [[2]]$user$genotypes[[1]]$genotype_id
#&gt; [1] 5001
#&gt;
#&gt; [[2]]$user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;CC&quot;
#&gt;
#&gt;
#&gt;
#&gt;
#&gt;
#&gt; [[3]]
#&gt; [[3]]$snp
#&gt; [[3]]$snp$name
#&gt; [1] &quot;rs7412&quot;
#&gt;
#&gt; [[3]]$snp$chromosome
#&gt; [1] &quot;19&quot;
#&gt;
#&gt; [[3]]$snp$position
#&gt; [1] &quot;44908822&quot;
#&gt;
#&gt;
#&gt; [[3]]$user
#&gt; [[3]]$user$name
#&gt; [1] &quot;mdmoore07&quot;
#&gt;
#&gt; [[3]]$user$id
#&gt; [1] 6597
#&gt;
#&gt; [[3]]$user$genotypes
#&gt; [[3]]$user$genotypes[[1]]
#&gt; [[3]]$user$genotypes[[1]]$genotype_id
#&gt; [1] 5000
#&gt;
#&gt; [[3]]$user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;--&quot;
</code></pre>

<pre><code class="language-r">allgensnp('rs7412', df=TRUE)[1:10,]
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/rs7412.json
</code></pre>

<pre><code>#&gt;    snp_name snp_chromosome snp_position      user_name user_id genotype_id
#&gt; 1    rs7412             19     44908822      Luke Reid    6621        5023
#&gt; 2    rs7412             19     44908822      Ganesha18    6598        5001
#&gt; 3    rs7412             19     44908822      mdmoore07    6597        5000
#&gt; 4    rs7412             19     44908822 KevinHendricks    6590        4988
#&gt; 5    rs7412             19     44908822       jlafount    6589        4987
#&gt; 6    rs7412             19     44908822          fawaz    6587        4985
#&gt; 7    rs7412             19     44908822        tzankel    6582        4980
#&gt; 8    rs7412             19     44908822        David53    3997        2691
#&gt; 9    rs7412             19     44908822        marsha     6577        4974
#&gt; 10   rs7412             19     44908822      wbworkman    6574        4973
#&gt;    genotype
#&gt; 1        CC
#&gt; 2        CC
#&gt; 3        --
#&gt; 4        CC
#&gt; 5        CC
#&gt; 6        CC
#&gt; 7        CT
#&gt; 8        CT
#&gt; 9        CC
#&gt; 10       CC
</code></pre>

<h4 id="all-phenotypes">All Phenotypes</h4>

<p>Get all phenotypes, their variations, and how many users have data available for a given phenotype</p>

<p>Get all data</p>

<pre><code class="language-r">allphenotypes(df = TRUE)[1:10,]
</code></pre>

<pre><code>#&gt;    id characteristic    known_variations number_of_users
#&gt; 1   1      Eye color               Brown            1168
#&gt; 2   1      Eye color         Brown-green            1168
#&gt; 3   1      Eye color          Blue-green            1168
#&gt; 4   1      Eye color           Blue-grey            1168
#&gt; 5   1      Eye color               Green            1168
#&gt; 6   1      Eye color                Blue            1168
#&gt; 7   1      Eye color Hazel (brown/green)            1168
#&gt; 8   1      Eye color               Hazel            1168
#&gt; 9   1      Eye color               Mixed            1168
#&gt; 10  1      Eye color           Gray-blue            1168
</code></pre>

<p>Output a list, then call the characterisitc of interest by &lsquo;id&rsquo; or &lsquo;characteristic&rsquo;</p>

<pre><code class="language-r">datalist &lt;- allphenotypes()
</code></pre>

<p>Get a list of all characteristics you can call</p>

<pre><code class="language-r">names(datalist)[1:10]
</code></pre>

<pre><code>#&gt;  [1] &quot;Eye color&quot;                        &quot;Lactose intolerance&quot;
#&gt;  [3] &quot;Handedness&quot;                       &quot;white skin&quot;
#&gt;  [5] &quot;Ability to find a bug in openSNP&quot; &quot;Beard Color&quot;
#&gt;  [7] &quot;Hair Color&quot;                       &quot;Ability to Tan&quot;
#&gt;  [9] &quot;Height&quot;                           &quot;Hair Type&quot;
</code></pre>

<p>Get data.frame for <em>ADHD</em></p>

<pre><code class="language-r">datalist[[&quot;ADHD&quot;]]
</code></pre>

<pre><code>#&gt;    id characteristic                            known_variations
#&gt; 1  29           ADHD                                       False
#&gt; 2  29           ADHD                                        True
#&gt; 3  29           ADHD              Undiagnosed, but probably true
#&gt; 4  29           ADHD                                          No
#&gt; 5  29           ADHD                                         Yes
#&gt; 6  29           ADHD                               Not diagnosed
#&gt; 7  29           ADHD Diagnosed as not having but with some signs
#&gt; 8  29           ADHD                                 Mthfr c677t
#&gt; 9  29           ADHD                                   Rs1801260
#&gt; 10 29           ADHD                                 Adult onset
#&gt;    number_of_users
#&gt; 1              261
#&gt; 2              261
#&gt; 3              261
#&gt; 4              261
#&gt; 5              261
#&gt; 6              261
#&gt; 7              261
#&gt; 8              261
#&gt; 9              261
#&gt; 10             261
</code></pre>

<p>Get data.frame for <em>mouth size</em> and <em>SAT Writing</em></p>

<pre><code class="language-r">datalist[c(&quot;mouth size&quot;,&quot;SAT Writing&quot;)]
</code></pre>

<pre><code>#&gt; $`mouth size`
#&gt;    id characteristic     known_variations number_of_users
#&gt; 1 120     mouth size               Medium             173
#&gt; 2 120     mouth size                Small             173
#&gt; 3 120     mouth size                Large             173
#&gt; 4 120     mouth size Slightly wide mouth              173
#&gt;
#&gt; $`SAT Writing`
#&gt;    id characteristic
#&gt; 1  41    SAT Writing
#&gt; 2  41    SAT Writing
#&gt; 3  41    SAT Writing
#&gt; 4  41    SAT Writing
#&gt; 5  41    SAT Writing
#&gt; 6  41    SAT Writing
#&gt; 7  41    SAT Writing
#&gt; 8  41    SAT Writing
#&gt; 9  41    SAT Writing
#&gt; 10 41    SAT Writing
#&gt; 11 41    SAT Writing
#&gt; 12 41    SAT Writing
#&gt; 13 41    SAT Writing
#&gt; 14 41    SAT Writing
#&gt; 15 41    SAT Writing
#&gt;                                           known_variations number_of_users
#&gt; 1                                                      750              92
#&gt; 2                                       Tested before 2005              92
#&gt; 3                                                      800              92
#&gt; 4                                      Country with no sat              92
#&gt; 5                                                      N/a              92
#&gt; 6                                  Never &amp; have ba &amp; above              92
#&gt; 7                                                      720              92
#&gt; 8                          Did well - don't remember score              92
#&gt; 9                                                      511              92
#&gt; 10                                                     700              92
#&gt; 11                                                     760              92
#&gt; 12                                                     780              92
#&gt; 13 Not part of sat when i took test in august 1967 at uiuc              92
#&gt; 14                                 Not part of sat in 1961              92
#&gt; 15                                                     620              92
</code></pre>

<h4 id="annotations">Annotations</h4>

<p>Get just the metadata</p>

<pre><code class="language-r">annotations(snp = 'rs7903146', output = 'metadata')
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/json/annotation/rs7903146.json
</code></pre>

<pre><code>#&gt;          .id        V1
#&gt; 1       name rs7903146
#&gt; 2 chromosome        10
#&gt; 3   position 112998590
</code></pre>

<p>Just from PLOS journals</p>

<pre><code class="language-r">annotations(snp = 'rs7903146', output = 'plos')[c(1:2),]
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/json/annotation/rs7903146.json
</code></pre>

<pre><code>#&gt;                   author
#&gt; 1        Maggie C. Y. Ng
#&gt; 2 Andr√© Gustavo P. Sousa
#&gt;                                                                                                                                      title
#&gt; 1 Meta-Analysis of Genome-Wide Association Studies in African Americans Provides Insights into the Genetic Architecture of Type 2 Diabetes
#&gt; 2                                  Genetic Variants of Diabetes Risk and Incident Cardiovascular Events in Chronic Coronary Artery Disease
#&gt;           publication_date number_of_readers
#&gt; 1 2014-08-07T00:00:00.000Z              7783
#&gt; 2 2011-01-20T00:00:00.000Z              2080
#&gt;                                              url
#&gt; 1 http://dx.doi.org/10.1371/journal.pgen.1004517
#&gt; 2 http://dx.doi.org/10.1371/journal.pone.0016341
#&gt;                            doi
#&gt; 1 10.1371/journal.pgen.1004517
#&gt; 2 10.1371/journal.pone.0016341
</code></pre>

<p>Just from SNPedia</p>

<pre><code class="language-r">annotations(snp = 'rs7903146', output = 'snpedia')
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/json/annotation/rs7903146.json
</code></pre>

<pre><code>#&gt;                                               url
#&gt; 1 http://www.snpedia.com/index.php/Rs7903146(C;C)
#&gt; 2 http://www.snpedia.com/index.php/Rs7903146(C;T)
#&gt; 3 http://www.snpedia.com/index.php/Rs7903146(T;T)
#&gt;                                                            summary
#&gt; 1 Normal (lower) risk of Type 2 Diabetes and Gestational Diabetes.
#&gt; 2     1.4x increased risk for diabetes (and perhaps colon cancer).
#&gt; 3                            2x increased risk for Type-2 diabetes
</code></pre>

<p>Get all annotations</p>

<pre><code class="language-r">annotations(snp = 'rs7903146', output = 'all')[1:5,]
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/json/annotation/rs7903146.json
</code></pre>

<pre><code>#&gt;        .id              author
#&gt; 1 mendeley           T E Meyer
#&gt; 2 mendeley      Camilla Cervin
#&gt; 3 mendeley Nicholette D Palmer
#&gt; 4 mendeley      Ashis K Mondal
#&gt; 5 mendeley        Julian Munoz
#&gt;                                                                                                                                title
#&gt; 1                                                Diabetes genes and prostate cancer in the Atherosclerosis Risk in Communities study
#&gt; 2                                                        Diabetes in Adults , Type 1 Diabetes , and Type 2 Diabetes GENETICS OF LADA
#&gt; 3                                Association of TCF7L2 gene polymorphisms with reduced acute insulin response in Hispanic Americans.
#&gt; 4                  Genotype and tissue-specific effects on alternative splicing of the transcription factor 7-like 2 gene in humans.
#&gt; 5 Polymorphism in the transcription factor 7-like 2 (TCF7L2) gene is associated with reduced insulin secretion in nondiabetic women.
#&gt;   publication_year number_of_readers open_access
#&gt; 1             2010                 3        TRUE
#&gt; 2             2008                 2       FALSE
#&gt; 3             2008                 8       FALSE
#&gt; 4             2010                13        TRUE
#&gt; 5             2006                10        TRUE
#&gt;                                                                                                                                      url
#&gt; 1                              http://www.mendeley.com/research/diabetes-genes-prostate-cancer-atherosclerosis-risk-communities-study-4/
#&gt; 2                                        http://www.mendeley.com/research/diabetes-adults-type-1-diabetes-type-2-diabetes-genetics-lada/
#&gt; 3              http://www.mendeley.com/research/association-tcf7l2-gene-polymorphisms-reduced-acute-insulin-response-hispanic-americans/
#&gt; 4        http://www.mendeley.com/research/genotype-tissuespecific-effects-alternative-splicing-transcription-factor-7like-2-gene-humans/
#&gt; 5 http://www.mendeley.com/research/polymorphism-transcription-factor-7like-2-tcf7l2-gene-associated-reduced-insulin-secretion-nondiabet/
#&gt;                                              doi publication_date summary
#&gt; 1 19/2/558 [pii]\\r10.1158/1055-9965.EPI-09-0902             &lt;NA&gt;    &lt;NA&gt;
#&gt; 2                         10.2337/db07-0299.Leif             &lt;NA&gt;    &lt;NA&gt;
#&gt; 3                           10.1210/jc.2007-1225             &lt;NA&gt;    &lt;NA&gt;
#&gt; 4                           10.1210/jc.2009-2064             &lt;NA&gt;    &lt;NA&gt;
#&gt; 5                              10.2337/db06-0574             &lt;NA&gt;    &lt;NA&gt;
#&gt;   first_author pubmed_link journal trait pvalue pvalue_description
#&gt; 1         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt; 2         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt; 3         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt; 4         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt; 5         &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;     NA               &lt;NA&gt;
#&gt;   confidence_interval
#&gt; 1                &lt;NA&gt;
#&gt; 2                &lt;NA&gt;
#&gt; 3                &lt;NA&gt;
#&gt; 4                &lt;NA&gt;
#&gt; 5                &lt;NA&gt;
</code></pre>

<h4 id="download">Download</h4>

<p>Download genotype data for a user from 23andme or other repo. (not evaluated in this example)</p>

<pre><code class="language-r">data &lt;- users(df=TRUE)
head( data[[1]] )
fetch_genotypes(url = data[[1]][1,&quot;genotypes.download_url&quot;], rows=15)
</code></pre>

<h4 id="genotype-user-data">Genotype user data</h4>

<p>Genotype data for one or multiple users</p>

<pre><code class="language-r">genotypes(snp='rs9939609', userid=1)
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/json/rs9939609/1.json
</code></pre>

<pre><code>#&gt; $snp
#&gt; $snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt;
#&gt; $snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt;
#&gt; $snp$position
#&gt; [1] &quot;53786615&quot;
#&gt;
#&gt;
#&gt; $user
#&gt; $user$name
#&gt; [1] &quot;Bastian Greshake Tzovaras&quot;
#&gt;
#&gt; $user$id
#&gt; [1] 1
#&gt;
#&gt; $user$genotypes
#&gt; $user$genotypes[[1]]
#&gt; $user$genotypes[[1]]$genotype_id
#&gt; [1] 9
#&gt;
#&gt; $user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;AT&quot;
</code></pre>

<pre><code class="language-r">genotypes('rs9939609', userid='1,6,8', df=TRUE)
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/json/rs9939609/1,6,8.json
</code></pre>

<pre><code>#&gt;    snp_name snp_chromosome snp_position                 user_name user_id
#&gt; 1 rs9939609             16     53786615 Bastian Greshake Tzovaras       1
#&gt; 2 rs9939609             16     53786615              Nash Parovoz       6
#&gt; 3 rs9939609             16     53786615         Samantha B. Clark       8
#&gt;   genotype_id genotype
#&gt; 1           9       AT
#&gt; 2           5       AT
#&gt; 3           2       TT
</code></pre>

<pre><code class="language-r">genotypes('rs9939609', userid='1-2', df=FALSE)
</code></pre>

<pre><code>#&gt; https://opensnp.org/snps/json/rs9939609/1-2.json
</code></pre>

<pre><code>#&gt; [[1]]
#&gt; [[1]]$snp
#&gt; [[1]]$snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt;
#&gt; [[1]]$snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt;
#&gt; [[1]]$snp$position
#&gt; [1] &quot;53786615&quot;
#&gt;
#&gt;
#&gt; [[1]]$user
#&gt; [[1]]$user$name
#&gt; [1] &quot;Bastian Greshake Tzovaras&quot;
#&gt;
#&gt; [[1]]$user$id
#&gt; [1] 1
#&gt;
#&gt; [[1]]$user$genotypes
#&gt; [[1]]$user$genotypes[[1]]
#&gt; [[1]]$user$genotypes[[1]]$genotype_id
#&gt; [1] 9
#&gt;
#&gt; [[1]]$user$genotypes[[1]]$local_genotype
#&gt; [1] &quot;AT&quot;
#&gt;
#&gt;
#&gt;
#&gt;
#&gt;
#&gt; [[2]]
#&gt; [[2]]$snp
#&gt; [[2]]$snp$name
#&gt; [1] &quot;rs9939609&quot;
#&gt;
#&gt; [[2]]$snp$chromosome
#&gt; [1] &quot;16&quot;
#&gt;
#&gt; [[2]]$snp$position
#&gt; [1] &quot;53786615&quot;
#&gt;
#&gt;
#&gt; [[2]]$user
#&gt; [[2]]$user$name
#&gt; [1] &quot;Senficon&quot;
#&gt;
#&gt; [[2]]$user$id
#&gt; [1] 2
#&gt;
#&gt; [[2]]$user$genotypes
#&gt; list()
</code></pre>

<h4 id="phenotype-user-data">Phenotype user data</h4>

<p>Get phenotype data for one or multiple users</p>

<pre><code class="language-r">phenotypes(userid=1)$phenotypes[1:3]
</code></pre>

<pre><code>#&gt; https://opensnp.org/phenotypes/json/1.json
</code></pre>

<pre><code>#&gt; $syndactyly
#&gt; $syndactyly$phenotype_id
#&gt; [1] 372
#&gt;
#&gt; $syndactyly$variation
#&gt; [1] &quot;None&quot;
#&gt;
#&gt;
#&gt; $`Allergy to Hair Dye`
#&gt; $`Allergy to Hair Dye`$phenotype_id
#&gt; [1] 370
#&gt;
#&gt; $`Allergy to Hair Dye`$variation
#&gt; [1] &quot;None&quot;
#&gt;
#&gt;
#&gt; $`Do You Have Lucid Dreams?`
#&gt; $`Do You Have Lucid Dreams?`$phenotype_id
#&gt; [1] 328
#&gt;
#&gt; $`Do You Have Lucid Dreams?`$variation
#&gt; [1] &quot;No&quot;
</code></pre>

<pre><code class="language-r">phenotypes(userid='1,6,8', df=TRUE)[[1]][1:10,]
</code></pre>

<pre><code>#&gt; https://opensnp.org/phenotypes/json/1,6,8.json
</code></pre>

<pre><code>#&gt;                       phenotype phenotypeID  variation
#&gt; 1                    syndactyly         372       None
#&gt; 2           Allergy to Hair Dye         370       None
#&gt; 3     Do You Have Lucid Dreams?         328         No
#&gt; 4            Multiple Sclerosis         215       None
#&gt; 5      Can you smell cut-grass?         168        Yes
#&gt; 6  Daily Sleep Duration (hours)         269        4-6
#&gt; 7                    blood type         290         A+
#&gt; 8             Cocaine addiction         486         No
#&gt; 9          macular degeneration         542         No
#&gt; 10                         Diet         533 Vegetarian
</code></pre>

<pre><code class="language-r">out &lt;- phenotypes(userid='1-8', df=TRUE)
</code></pre>

<pre><code>#&gt; https://opensnp.org/phenotypes/json/1-8.json
</code></pre>

<pre><code class="language-r">lapply(out, head)
</code></pre>

<pre><code>#&gt; $`Bastian Greshake Tzovaras`
#&gt;                      phenotype phenotypeID variation
#&gt; 1                   syndactyly         372      None
#&gt; 2          Allergy to Hair Dye         370      None
#&gt; 3    Do You Have Lucid Dreams?         328        No
#&gt; 4           Multiple Sclerosis         215      None
#&gt; 5     Can you smell cut-grass?         168       Yes
#&gt; 6 Daily Sleep Duration (hours)         269       4-6
#&gt;
#&gt; $Senficon
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt;
#&gt; $`no info on user_3`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt;
#&gt; $`no info on user_4`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt;
#&gt; $`no info on user_5`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt;
#&gt; $`Nash Parovoz`
#&gt;                          phenotype phenotypeID        variation
#&gt; 1                       Handedness           3     right-handed
#&gt; 2                        Eye color           1            brown
#&gt; 3                       white skin           4        Caucasian
#&gt; 4              Lactose intolerance           2 lactose-tolerant
#&gt; 5 Ability to find a bug in openSNP           5   extremely high
#&gt; 6           Number of wisdom teeth          57                4
#&gt;
#&gt; $`no info on user_7`
#&gt;   phenotype phenotypeID variation
#&gt; 1   no data     no data   no data
#&gt;
#&gt; $`Samantha B. Clark`
#&gt;                             phenotype phenotypeID           variation
#&gt; 1                            Gambling         539                  No
#&gt; 2                 Caffeine dependence         538                  No
#&gt; 3            Dietary supplements used         534                 b12
#&gt; 4                                Diet         533 Vegan / plant-based
#&gt; 5                   Tooth sensitivity         532         Sweet, cold
#&gt; 6 OCD - Obsessive-Compulsive Disorder         555                  No
</code></pre>

<h4 id="all-known-variations">All known variations</h4>

<p>Get all known variations and all users sharing that phenotype for one phenotype(-ID).</p>

<pre><code class="language-r">phenotypes_byid(phenotypeid=12, return_ = 'desc')
</code></pre>

<pre><code>#&gt; https://opensnp.org/phenotypes/json/variations/12.json
</code></pre>

<pre><code>#&gt; $id
#&gt; [1] 12
#&gt;
#&gt; $characteristic
#&gt; [1] &quot;Beard Color&quot;
#&gt;
#&gt; $description
#&gt; [1] &quot;coloration of facial hair&quot;
</code></pre>

<pre><code class="language-r">phenotypes_byid(phenotypeid=12, return_ = 'knownvars')
</code></pre>

<pre><code>#&gt; https://opensnp.org/phenotypes/json/variations/12.json
</code></pre>

<pre><code>#&gt; $known_variations
#&gt; $known_variations[[1]]
#&gt; [1] &quot;Red&quot;
#&gt;
#&gt; $known_variations[[2]]
#&gt; [1] &quot;Blonde&quot;
#&gt;
#&gt; $known_variations[[3]]
#&gt; [1] &quot;Red-brown&quot;
#&gt;
#&gt; $known_variations[[4]]
#&gt; [1] &quot;Red-blonde-brown-black(in diferent parts i have different color,for example near the lips blond-red&quot;
#&gt;
#&gt; $known_variations[[5]]
#&gt; [1] &quot;No beard-female&quot;
#&gt;
#&gt; $known_variations[[6]]
#&gt; [1] &quot;Brown-black&quot;
#&gt;
#&gt; $known_variations[[7]]
#&gt; [1] &quot;Blonde-brown&quot;
#&gt;
#&gt; $known_variations[[8]]
#&gt; [1] &quot;Black&quot;
#&gt;
#&gt; $known_variations[[9]]
#&gt; [1] &quot;Dark brown with minor blondish-red&quot;
#&gt;
#&gt; $known_variations[[10]]
#&gt; [1] &quot;Brown-grey&quot;
#&gt;
#&gt; $known_variations[[11]]
#&gt; [1] &quot;Red-blonde-brown-black&quot;
#&gt;
#&gt; $known_variations[[12]]
#&gt; [1] &quot;Blond-brown&quot;
#&gt;
#&gt; $known_variations[[13]]
#&gt; [1] &quot;Brown, some red&quot;
#&gt;
#&gt; $known_variations[[14]]
#&gt; [1] &quot;Brown&quot;
#&gt;
#&gt; $known_variations[[15]]
#&gt; [1] &quot;Brown-gray&quot;
#&gt;
#&gt; $known_variations[[16]]
#&gt; [1] &quot;Never had a beard&quot;
#&gt;
#&gt; $known_variations[[17]]
#&gt; [1] &quot;I'm a woman&quot;
#&gt;
#&gt; $known_variations[[18]]
#&gt; [1] &quot;Black-brown-blonde&quot;
#&gt;
#&gt; $known_variations[[19]]
#&gt; [1] &quot;Was red-brown now mixed with gray,&quot;
#&gt;
#&gt; $known_variations[[20]]
#&gt; [1] &quot;Red-blonde-brown&quot;
#&gt;
#&gt; $known_variations[[21]]
#&gt; [1] &quot;Dark brown w/few blonde &amp; red hairs&quot;
#&gt;
#&gt; $known_variations[[22]]
#&gt; [1] &quot;Dark blonde with red and light blonde on goatee area.&quot;
#&gt;
#&gt; $known_variations[[23]]
#&gt; [1] &quot;Black with few red hairs&quot;
#&gt;
#&gt; $known_variations[[24]]
#&gt; [1] &quot;Black, graying&quot;
#&gt;
#&gt; $known_variations[[25]]
#&gt; [1] &quot;Red, moustache still is, beard mostly white&quot;
#&gt;
#&gt; $known_variations[[26]]
#&gt; [1] &quot;Blonde/brown-some black-and red on chin-all starting to gray&quot;
#&gt;
#&gt; $known_variations[[27]]
#&gt; [1] &quot;Dark brown&quot;
#&gt;
#&gt; $known_variations[[28]]
#&gt; [1] &quot;Every possible color, most hair shafts have more than one color at different points along the shaft&quot;
</code></pre>

<pre><code class="language-r">phenotypes_byid(phenotypeid=12, return_ = 'users')[1:10,]
</code></pre>

<pre><code>#&gt; https://opensnp.org/phenotypes/json/variations/12.json
</code></pre>

<pre><code>#&gt;    user_id
#&gt; 1       22
#&gt; 2        1
#&gt; 3       26
#&gt; 4       10
#&gt; 5       14
#&gt; 6       42
#&gt; 7       45
#&gt; 8       16
#&gt; 9        8
#&gt; 10     661
#&gt;                                                                                              variation
#&gt; 1                                                                                                  Red
#&gt; 2                                                                                               Blonde
#&gt; 3                                                                                            red-brown
#&gt; 4  Red-Blonde-Brown-Black(in diferent parts i have different color,for example near the lips blond-red
#&gt; 5                                                                                      No beard-female
#&gt; 6                                                                                          Brown-black
#&gt; 7  Red-Blonde-Brown-Black(in diferent parts i have different color,for example near the lips blond-red
#&gt; 8                                                                                         blonde-brown
#&gt; 9                                                                                      No beard-female
#&gt; 10                                                                                         Brown-black
</code></pre>

<h4 id="opensnp-users">OpenSNP users</h4>

<pre><code class="language-r">data &lt;- users(df=FALSE)
data[1:2]
</code></pre>

<pre><code>#&gt; [[1]]
#&gt; [[1]]$name
#&gt; [1] &quot;gigatwo&quot;
#&gt;
#&gt; [[1]]$id
#&gt; [1] 31
#&gt;
#&gt; [[1]]$genotypes
#&gt; list()
#&gt;
#&gt;
#&gt; [[2]]
#&gt; [[2]]$name
#&gt; [1] &quot;Anu Acharya&quot;
#&gt;
#&gt; [[2]]$id
#&gt; [1] 385
#&gt;
#&gt; [[2]]$genotypes
#&gt; list()
</code></pre>

<h3 id="ncbi-snp-data">NCBI SNP data</h3>

<h4 id="dbsnp">dbSNP</h4>

<p>Query NCBI&rsquo;s dbSNP for information on a set of SNPs</p>

<p>An example with both merged SNPs, non-SNV SNPs, regular SNPs, SNPs not found, microsatellite</p>

<pre><code class="language-r">snps &lt;- c(&quot;rs332&quot;, &quot;rs420358&quot;, &quot;rs1837253&quot;, &quot;rs1209415715&quot;, &quot;rs111068718&quot;)
NCBI_snp_query(snps)
</code></pre>

<pre><code>#&gt; Warning: use ncbi_snp_query instead - NCBI_snp_query removed in next
#&gt; version
</code></pre>

<pre><code>#&gt; Warning: The following rsIds had no information available on NCBI:
#&gt;   rs1209415715, rs111068718
</code></pre>

<pre><code>#&gt; Warning: rs332 has been merged into rs121909001
</code></pre>

<pre><code>#&gt;       Query Chromosome      Marker  Class Gene Alleles Major Minor    MAF
#&gt; 1     rs332          7 rs121909001 in-del CFTR   -/TTT  &lt;NA&gt;  &lt;NA&gt;     NA
#&gt; 2  rs420358          1    rs420358    snp &lt;NA&gt;     G,T     G     T     NA
#&gt; 3 rs1837253          5   rs1837253    snp &lt;NA&gt;     C/T     C     T 0.3822
#&gt;          BP AncestralAllele
#&gt; 1 117559593            &lt;NA&gt;
#&gt; 2  40341239     T,T,T,T,T,T
#&gt; 3 111066174     T,T,T,T,T,T
</code></pre>

<h3 id="citing">Citing</h3>

<p>To cite <code>rsnps</code> in publications use:</p>

<p><br></p>

<blockquote>
<p>Scott Chamberlain, Kevin Ushey and Hao Zhu (2016). rsnps: Get &lsquo;SNP&rsquo;
  (&lsquo;Single-Nucleotide&rsquo; &lsquo;Polymorphism&rsquo;) Data on the Web. R package
  version 0.2.0. <a href="https://CRAN.R-project.org/package=rsnps">https://CRAN.R-project.org/package=rsnps</a></p>
</blockquote>

<h3 id="license-and-bugs">License and bugs</h3>

<ul>
<li>License: <a href="http://opensource.org/licenses/MIT">MIT</a></li>
<li>Report bugs at <a href="https://github.com/ropensci/rsnps/issues?state=open">our Github repo for rsnps</a></li>
</ul>

<p><a href="#top">Back to top</a></p>

                    </article>
                </div>

                </div>
            </div>
        </div>
    </article>


    <div id="footer">
        <div class="container">
            <div class="row start">
                <div class="col-1 col-1-3 top-8">
                    <img src="/img/icon_short_white.svg" />
                </div>

                <div class="col-9 col-offset-2 top-10 bottom-8">
                    <div class="row between">
                        <div class="col-2 col-1-3">
                            <a href="http://github.com/ropensci" target="_blank"><div class="icon icon-github"></div></a>
                            <a href="http://twitter.com/ropensci" target="_blank"><div class="icon icon-twitter"></div></a>
                            <a href="http://vimeo.com/ropensci" target="_blank"><div class="icon icon-vimeo"></div></a>
                        </div>
                    </div>
                    <div class="row">
                        <div class="col-2 col-1-3 top-8 bottom-8">
                            <ul>
                                <h5 class="bottom-2">Info</h5>
                                <li><a href="/about">Mission</a></li>
                                <li><a href="/about#team">Team</a></li>
                                <li><a href="/about#collaborators">Collaborators</a></li>
                                <li><a href="/careers">Careers</a></li>

                            </ul>
                        </div>

                        <div class="col-2 col-1-3 top-8 bottom-8">
                            <ul>
                                <h5 class="bottom-2">Work</h5>
                                <li><a href="/packages/">Packages</a></li>
                                <li><a href="/blog/">Blog</a></li>
                                <li><a href="/technotes/">Tech Notes</a></li>
                                <li><a href="/tutorials/">Tutorials</a></li>
                                <li><a href="/usecases/">Use Cases</a></li>
                                <li><a href="/resources/">More Resources</a></li>
                            </ul>
                        </div>
                        <div class="col-2 col-1-3 top-8 bottom-8">
                            <ul>
                                <h5 class="bottom-2">Participate</h5>
                                <li><a href="/contact.html">Contact us</a></li>
                                <li><a href="/community/">Community</a></li>
                                <li><a href="http://onboarding.ropensci.org/">Contribute software</a></li>
                                <li><a href="http://unconf17.ropensci.org/">Unconference</a></li>
                                <li><a href="/coc">Code of conduct</a></li>
                                <li><a href="/donate/">Donate</a></li>
                            </ul>
                        </div>
                        <div class="col-4 top-8 bottom-8">
                            <h5 class="bottom-2"></h5>

                            <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" target="_blank">NumFOCUS</a></p>
                            <a href="http://numfocus.org" target="_blank"><img src="img/numfocus.png"></a>
                            <br>

                        </div>
                    </div>
                </div>
            </div>
            <div class="row">
                <div class="col-12 top-8 bottom-9 divider">
                    <p class="top-9">Except where otherwise noted, content on this site is licensed under the <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank">CC-BY license</a>.
                    </p>
                </div>
            </div>
        </div>
    </div>




    
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.4/jquery.min.js"></script>
    

    
    <script type="text/javascript" src="/js/highlight.pack.js"></script>
    <script type="text/javascript">
        hljs.initHighlightingOnLoad();
    </script>



</body>
</html>
